pingouin/pairwise.py

```diff
@@ -1132,12 +1132,12 @@ def traverse(o, tree_types=(list, tuple)):
         assert all([c in keys for c in covar]), (
             'covar not in data, not num or single unique value.'
         )
-        # Assert that there is at least one combination
-        if stats.shape[0] == 0:
-            raise ValueError("No column combination found. Please make sure "
-                             "that the specified columns and covar exist in "
-                             "the dataframe, are numeric, and contain at "
-                             "least two unique values.")
+        # # Assert that there is at least one combination
+        # if stats.shape[0] == 0:
+        #     raise ValueError("No column combination found. Please make sure "
+        #                      "that the specified columns and covar exist in "
+        #                      "the dataframe, are numeric, and contain at "
+        #                      "least two unique values.")
```

Comment:
Is this still required? This just seems to reduce codecov, while line 1104 already checks for zero-length data.

Reply:
I think this was just to handle the case where you end up with an empty dataframe after `stats = stats[~stats[['X', 'Y']].isin(covar).any(1)]`, but if you find that it doesn't reduce coverage to remove it with your changes, then please feel free to remove it.

Reply:
Yep. In contrast: Coverage is reduced if it is not removed, since these lines won't be reached anymore (now shape[0] is always >0).